### ISSUE TITLE
cpu explodes on empty workspace

### ISSUE THREAD

**Issue by oberon2007** [olemartinorg/i3-alternating-layout#6]
I have conky running on the desktop and when I create an empty workspace cpu usage of process **python** goes up to 15-25%. As soon as I open at least one container, cpu goes back to normal.
The script seems to have a problem with detecting an empty workspace?

**Comment by oberon2007** [comment]
Other than that it's an awesome idea. :) I love your script and I have just packaged it for Manjaro Linux.
It would be great if you could fix that problem...

**Comment by Airblader** [comment]
This patch seems to fix it. @oberon2007 Maybe you can verify

````
diff --git a/alternating_layouts.py b/alternating_layouts.py
index a335c85..32c52b5 100755
--- a/alternating_layouts.py
+++ b/alternating_layouts.py
@@ -56,6 +56,7 @@ def main():
     """
     process = subprocess.Popen(
         ['xprop', '-root', '-spy'],
+        shell = True,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE
     )
```

It does happen because it busy loops as I suspected in the Manjaro-i3 forums. I'm not sure why this particular patch works because I haven't looked into the docs yet on why it makes `readline` behave differently, but I simply compared it to the proposed way of blocking reads found in articles.

**Comment by olemartinorg** [comment]
Thanks @Airblader! I just read the emails i got for this issue, and I haven't had time to research it yet. Hoping that your fix does the trick! 

Would you mind linking to the thread in the Manjaro-i3 forums, in case anyone else stumbles upon this and wants to get the full picture?

**Comment by Airblader** [comment]
Sure. The problem was described here: https://forum.manjaro.org/index.php?topic=29773.msg248539#msg248539

Let's wait for @oberon2007 to confirm the fix, though. :)

**Comment by oberon2007** [comment]
Thanks a lot @Airblader . Unfortunately that doesn't fix it. Actually it does weird things:
launching the script does nothing but change the mouse-cursor to a selection cross :laughing: Then invoking the script a 2nd time it works BUT cpu here goes to 25% and stays there  even when I open containers.

**Comment by Airblader** [comment]
This is so odd. I swear it worked for me earlier, but running it again I have exactly the same issues as you. Well, let's reopen then. Maybe @olemartinorg wants to look at it now then. Sorry about that. :)

**Comment by olemartinorg** [comment]
Allright, sorry about that. I debugged it now, and it seems that my script goes into an infinite loop when no window is open, as xprop events are produced every time my script changes the layout.

I applied a fix where it won't do anything if the same event is produced twice, and that seems to fix it. I'll until @oberon2007 can confirm the fix before closing the issue this time - i got a bit trigger happy last time.. ;-)
